lightning/src/chain/keysinterface.rs

```diff
@@ -1209,14 +1217,19 @@ impl KeysInterface for KeysManager {
 		ShutdownScript::new_p2wpkh_from_pubkey(self.shutdown_pubkey.clone())
 	}
 
-	fn get_channel_signer(&self, _inbound: bool, channel_value_satoshis: u64) -> Self::Signer {
-		let child_ix = self.channel_child_index.fetch_add(1, Ordering::AcqRel);
-		assert!(child_ix <= core::u32::MAX as usize);
+	fn generate_channel_keys_id(&self, user_channel_id: u128) -> [u8; 32] {
+		let child_idx = self.channel_child_index.fetch_add(1, Ordering::AcqRel);
+		assert!(child_idx <= core::u32::MAX as usize);
 		let mut id = [0; 32];
-		id[0..8].copy_from_slice(&byte_utils::be64_to_array(child_ix as u64));
-		id[8..16].copy_from_slice(&byte_utils::be64_to_array(self.starting_time_nanos as u64));
-		id[16..24].copy_from_slice(&byte_utils::be64_to_array(self.starting_time_secs));
-		self.derive_channel_keys(channel_value_satoshis, &id)
+		id[0..4].copy_from_slice(&byte_utils::be32_to_array(child_idx as u32));
+		id[4..8].copy_from_slice(&byte_utils::be32_to_array(self.starting_time_nanos));
+		id[8..16].copy_from_slice(&byte_utils::be64_to_array(self.starting_time_secs));
+		id[16..32].copy_from_slice(&byte_utils::be128_to_array(user_channel_id));
+		id
+	}
+
+	fn derive_channel_signer(&self, channel_value_satoshis: u64, channel_keys_id: [u8; 32]) -> Self::Signer {
+		self.derive_channel_keys(channel_value_satoshis, &channel_keys_id)
```

Comment:
Note, the comment around `derive_channel_keys` still mention `DynamicOutputP2WSH` which doesn't exist anymore as a type of internal output descriptor. If we can remove the stale ref.